plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftConnector.java

```diff
@@ -593,34 +799,43 @@ private String getVolumeName(String volume) {
         return "unknown-volume";
     }
 
-    private String waitAndRetrieveContainerID(String deploymentName) {
+    private String waitAndRetrieveContainerID(String deploymentName) throws IOException {
         for (int i = 0; i < OPENSHIFT_WAIT_POD_TIMEOUT; i++) {
             try {
                 Thread.sleep(OPENSHIFT_WAIT_POD_DELAY);
             } catch (InterruptedException ex) {
                 Thread.currentThread().interrupt();
             }
 
-            PodList pods = openShiftClient.pods()
+            List<Pod> pods = openShiftClient.pods()
                                 .inNamespace(this.openShiftCheProjectName)
-                                .list();
-
-            for (Pod p : pods.getItems()) {
-                String status = p.getStatus().getPhase();
-                String dc = p.getMetadata().getLabels().get(OPENSHIFT_DEPLOYMENT_LABEL);
-                if (OPENSHIFT_POD_STATUS_RUNNING.equals(status) && deploymentName.equals(dc)) {
-                    String containerID = p.getStatus().getContainerStatuses().get(0).getContainerID();
-                    String normalizedID = normalizeContainerID(containerID);
-                    openShiftClient.pods()
-                            .inNamespace(this.openShiftCheProjectName)
-                            .withName(p.getMetadata().getName())
-                            .edit()
-                            .editMetadata()
-                                .addToLabels(CHE_CONTAINER_IDENTIFIER_LABEL_KEY, getLabelFromContainerID(normalizedID))
-                            .endMetadata()
-                            .done();
-                    return normalizedID;
-                }
+                                .withLabel(OPENSHIFT_DEPLOYMENT_LABEL, deploymentName)
+                                .list()
+                                .getItems();
+
+            if (pods.size() < 1) {
+                throw new DockerException(String.format("Pod with deployment name %s not found",
```

Comment:
One reason I didn't do this is that OpenShiftException is unchecked, and so it wouldn't be obvious that it could be thrown when using the methods. DockerException is an IOException.